src/base/rss/rss_session.cpp

```diff
@@ -214,14 +214,20 @@ nonstd::expected<void, QString> Session::moveItem(Item *item, const QString &des
     Q_ASSERT(item);
     Q_ASSERT(item != rootFolder());
 
+    if (item->path() == destPath)
+        return {};
+
+    if (auto *folder = static_cast<Folder *>(item)) // if `item` is a `Folder`
+    {
+        if (destPath.startsWith(folder->path() + Item::PathSeparator))
+            return nonstd::make_unexpected(tr("Couldn't move folder into itself or its subfolder."));
```

Comment:
"Couldn't" sounds like it's allowed but for some reason it failed.
```suggestion
            return nonstd::make_unexpected(tr("Can't move a folder into itself or its subfolders."));
```

Reply:
"Movement of a folder into itself or its subfolders is disallowed." ??

Reply:
> "Couldn't" sounds like it's allowed but for some reason it failed.

Changed.

Reply:
> "Movement of a folder into itself or its subfolders is disallowed." ??

Yes. This is a conflict in the general case.